bin/templates/cordova/lib/build.js

```diff
@@ -105,11 +107,36 @@ function parseOpts (options, resolvedTarget, projectRoot) {
     }
 
     if (!ret.packageInfo) {
-        if (Object.keys(packageArgs).length > 0) {
+        // The following loop is to decide whether to print a warning about generating a signed archive
+        // We only want to produce a warning if they are using a config property that is related to signing, but
+        // missing the required properties for signing. We don't want to produce a warning if they are simply
+        // using a build property that isn't related to singing, such as --packageType
+        let shouldWarn = false;
+        const signingKeys = ['keystore', 'alias', 'storePassword', 'password', 'keystoreType'];
+        for (let key in packageArgs) {
```

Comment:
Coding nit: I think this for loop could be replaced by an array `reduce()` call. (An array `forEach()` call would also be a little cleaner than using a for loop.) I would be happy to make a quick proposal if you like.

Reply:
Yah probably. I'm a little bit old school with my javascript, I'm not used to using these new `Array` methods. I can update it to use `reduce` instead of a for loop.

Reply:
Actually I think `forEach` makes a bit more sense here, because `reduce` is more of an aggregation method and the `accumulator` value begins as the first item in the array, which I think is error prone in this case.

Reply:
I am also not so used to these new Array methods, actually did a little research before writing the coding nit. Next idea: what about using `concat()` to start with a bogus element, such as this (not tested, coded with help from MDN & other sites):

```js
const shouldWarn = [‘bogus’].concat(packageArgs)
    .reduce((shouldWarn, key) => (
        shouldWarn || signingKeys.indexOf(key) !== -1)
    ), false);
```

(I would probably add a quick comment to explain the less obvious aspects of this code.)

Or am I getting too fancy here?

Reply:
I personally think that's even more confusing. What I have currently is...

```
let shouldWarn = false;
const signingKeys = ['keystore', 'alias', 'storePassword', 'password', 'keystoreType'];

Object.keys(packageArgs).forEach((key) => {
    if (!shouldWarn && signingKeys.indexOf(key) > -1) {
        // If we enter this condition, we have a key used for signing a build,
        // but we are missing some required signing properties
        shouldWarn = true;
    }
});
```

However, I did miss that `reduce` have an initial parameter.

Keep in mind... that `packageArgs` is not an array, it is an object... which is why I opted for `for .. in` originally.

Reply:
I guess I shall keep the original for loop in this case then since we are handling an object and not an array?

Reply:
I would favor the idea you had to do `Object.keys(packageArgs).forEach(…)`, as I still think it looks nicer than the for loop. What I meant is that I should have accounted for using `Object.keys(…)` when I made my original comment on this code.

IMHO I would still favor using `reduce()`, together with `Object.keys()`, but not very strongly at this point. I think the way you did `Object.keys(packageArgs).forEach(…)` is the perfect compromise.

I also have no objection in case you decide to keep the original for loop, as I think this is primarily about coding style which we can always change later.

Reply:
I think the current code is perfectly fine and simple to understand.

Reply:
Keeping the original for the time being, pending discussion at #786